services/submission.js

```diff
@@ -0,0 +1,129 @@
+const {HistoryEventBuilder} = require("../domain/history-event");
+const {verifySession} = require("../verifier/user-info-verifier");
+const {getSortDirection} = require("../crdc-datahub-database-drivers/utility/mongodb-utility");
+const USER_CONSTANTS = require("../crdc-datahub-database-drivers/constants/user-constants");
+const ROLES = USER_CONSTANTS.USER.ROLES;
+const ALL_FILTER = "All";
+
+
+
+function listConditions(userID, userRole, aUserOrganization, params){
+    const validApplicationStatus = {status: {$in: [NEW, IN_PROGRESS, SUBMITTED, RELEASED, COMPLETED, ARCHIVED]}};
+    // Default conditons are:
+    // Make sure application has valid status
+    let conditions = [{$and: [validApplicationStatus]}];
+    
+    // Filter on organization and status
+    if (params.organization !== ALL_FILTER) {
+        conditions[0].$and.push({"organization.name": params.organization});
+    }
+    if (params.status !== ALL_FILTER) {
+        conditions[0].$and.push({status: params.status});
+    }
+    // List all applications if Fed Lead / Admin / Data Concierge / Data Curator
+    const listAllApplicationRoles = [ROLES.ADMIN, ROLES.FEDERAL_LEAD, ROLES.CURATOR, ROLES.DC_POC];
+    if (listAllApplicationRoles.includes(userRole)) {
+        return [{"$match": {"$or": conditions}}];
+    }
+    // If org owner, add condition to return all data submissions associated with their organization
+    if (userRole === ROLES.ORG_OWNER && aUserOrganization?.orgID) {
+        conditions[0].$and.push({"organization.name": aUserOrganization.orgName});
+        return [{"$match": {"$or": conditions}}];
+    }
+
+    // Add condition so submitters will only see their data submissions
+    // User's cant make submissions, so they will always have no submissions 
+    // search by applicant's user id
+    conditions[0].$and.push({"submitterID": userID});
+
+    return [{"$match": {"$or": conditions}}];
+};
+
+class Submission {
+    constructor(logCollection, submissionCollection, organizationService, userService) {
+        this.logCollection = logCollection;
+        this.submissionCollection = submissionCollection;
+        this.organizationService = organizationService;
+        this.userService = userService;
+    }
+
+    async createSubmission(params, context) {
+        // TODO: Add this requirement: Study abbreviation must have an approved application within user's organization
+        verifySession(context)
+            .verifyInitialized()
+            .verifyRole([ROLES.SUBMITTER, ROLES.ORG_OWNER]);
+        let submission = params.submission;
+        const userInfo = context.userInfo;
+        let newApplicationProperties = {
+            _id: v4(),
+            name: params.name,
+            submitterID: userInfo._id,
+            submitterName: formatApplicantName(userInfo),
+            organization: {_id: userInfo.organization.orgID, name: userInfo.organization.orgName},
```

Comment:
On my local machine, organization.orgID is always null. Could you double check?

Reply:
I believe that this is working with Alec's newest changes to user and organization. OrgID should be auto-populated inside the backend by user.js. This may be an issue with your local environment